generator/src/generator/namespace.rs

```diff
@@ -3522,20 +3675,32 @@ impl<'ns, 'c> NamespaceGenerator<'ns, 'c> {
                 _ => unreachable!(),
             };
             if !self.enum_has_repeated_values(&enum_def) {
-                outln!(out, "let {} = {}.try_into()?;", var_name, var_name);
+                let type_type = type_.type_.def.get().unwrap();
+                if let xcbdefs::TypeRef::BuiltIn(xcbdefs::BuiltInType::Bool) = type_type {
+                    // Cast this to a u8 before calling try_into, because there's
+                    // no TryFrom<bool> implementation.
+                    outln!(out, "let {var} = {var} as u8;", var = var_name);
+                }
+                outln!(out, "let {var} = {var}.try_into()?;", var = var_name);
```

Comment:
Would it make sense to actually implement `TryFrom<bool>` for enums?

Reply:
I think I asked the same thing. The answer is that this makes sense for `GrabOwner`, but not really for other affected enums: https://github.com/psychon/x11rb/pull/452#discussion_r430901102

Reply:
Can you change it to something like this?
```suggestion
                let type_type = type_.type_.def.get().unwrap();
                if let xcbdefs::TypeRef::BuiltIn(xcbdefs::BuiltInType::Bool) = type_type {
                    // Cast this to a u8 before calling try_into, because there's
                    // no TryFrom<bool> implementation.
                    outln!(out, "let {var} = u8::from({var}).try_into()?;", var = var_name);
                } else {
                    outln!(out, "let {var} = {var}.try_into()?;", var = var_name);
                }
```

Reply:
Is the point of this just to save that one line with the `bool as u8` conversion or is there something else I don't see?

Reply:
Yes, just to save the line.

Reply:
Yeah I think this would either a) need more data in the XML or b) we'd have to connect the use of the enum to the enum_def during the parsing phase.

Reply:
Sorry the previous comment responding to something else. This is resolved now.